pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -622,21 +640,34 @@ func (exp *FlowExporter) findFlowType(conn flowexporter.Connection) uint8 {
 	}
 
 	if exp.nodeRouteController == nil {
-		klog.V(4).InfoS("Can't find flowType without nodeRouteController")
-		return 0
+		if klog.V(5).Enabled() {
+			klog.InfoS("Can't find flowType without nodeRouteController")
+		}
+		return flowTypeUnknown
+	}
+	srcIsPod, srcIsGw := exp.nodeRouteController.LookupIPInPodSubnets(conn.FlowKey.SourceAddress)
+	dstIsPod, dstIsGw := exp.nodeRouteController.LookupIPInPodSubnets(conn.FlowKey.DestinationAddress)
+	if srcIsGw || dstIsGw {
+		// This matches what we do in filterAntreaConns but is more general as we consider
+		// remote gateways as well.
+		if klog.V(5).Enabled() {
+			klog.InfoS("Flows where the source or destination IP is a gateway IP will not be exported")
+		}
+		return flowTypeUnsupported
 	}
-	if exp.nodeRouteController.IPInPodSubnets(conn.FlowKey.SourceAddress.AsSlice()) {
-		if conn.Mark&openflow.ServiceCTMark.GetRange().ToNXRange().ToUint32Mask() == openflow.ServiceCTMark.GetValue() || exp.nodeRouteController.IPInPodSubnets(conn.FlowKey.DestinationAddress.AsSlice()) {
```

Comment:
@heanlan @yuntanghsu could you take a look at this change?
I tried to look through the Git history, but could not find the rationale for this check (`conn.Mark&openflow.ServiceCTMark.GetRange().ToNXRange().ToUint32Mask() == openflow.ServiceCTMark.GetValue()`) and there was no comment in the code. Why are we checking if the flow is a "Service flow"? it feels to me like the VIP does not matter when classifying the flow. Only the actual destination matters (is it a Pod and if yes, is it a local Pod or not?).

Reply:
I think this one might be copied and paste from deny_connections where we used this condition to determine whether we need to fill the service information?
I cannot remember it clearly.
But if it is a service flow, will LookupIPInPodSubnets return true for isPod? Because I think if a flow is a service flow then it must be FlowTypeInterNode or FlowTypeIntraNode?

Reply:
For Service traffic, `conn.FlowKey.DestinationAddress` will always be the resolved Endpoint IP, not the ClusterIP, even at the source Node. This is ensured by using `conn.TupleReply.IP.SourceAddress`:
https://github.com/antrea-io/antrea/blob/b3c40b07e68041d1e6773ed7b449341671480f7b/pkg/agent/flowexporter/connections/conntrack_linux.go#L136-L142

So yes `LookupIPInPodSubnets` will return true for Pod-to-Service traffic, both at the source Node and destination Node (incl. if these 2 Nodes are the same).

Reply:
All the tests are passing, so I have to assume that the new logic is valid...